src/compiler/program.ts

```diff
@@ -1146,8 +1146,10 @@ namespace ts {
                 const typeChecker = getDiagnosticsProducingTypeChecker();
 
                 Debug.assert(!!sourceFile.bindDiagnostics);
-                // For JavaScript files, we don't want to report semantic errors unless explicitly requested.
-                const includeBindAndCheckDiagnostics = !isSourceFileJavaScript(sourceFile) || isCheckJsEnabledForFile(sourceFile, options);
+
+                // By default, only type-check .ts, .tsx, and 'External' files (external files are added by plugins)
+                const includeBindAndCheckDiagnostics = sourceFile.scriptKind === ScriptKind.TS || sourceFile.scriptKind === ScriptKind.TSX ||
+                    sourceFile.scriptKind === ScriptKind.External || isCheckJsEnabledForFile(sourceFile, options);

```

Comment:
As an aside, I get confused every time I see a reference to "external" source files.  It sounds like they relate to external projects.  Maybe "plugin" or "extension" would be clearer?

Reply:
You're right, that is an aside ;-)

For this quick & urgent fix, I don't intend to do a large refactoring across both managed and script side where this terminology is used. Maybe a suggestion for later though.